src/app/story/directives/status.component.ts

```diff
@@ -0,0 +1,181 @@
+import { Component, Input, DoCheck } from '@angular/core';
+import { Router } from '@angular/router';
+import { ModalService } from '../../core';
+import { StoryModel } from '../../shared';
+
+@Component({
+  selector: 'publish-story-status',
+  styleUrls: ['status.component.css'],
+  template: `
+    <h1>Publish</h1>
+    <dl>
+
+      <dt>Status</dt>
+      <dd>
+        <span [class]="statusClass">{{statusText}}</span>
+        <template [ngIf]="isPublished">
+          <button *ngIf="editStatus" class="btn-link edit-status" (click)="toggleEdit()">Hide</button>
+          <button *ngIf="!editStatus" class="btn-link edit-status" (click)="toggleEdit()">Edit</button>
+          <publish-button *ngIf="editStatus" [model]="story" visible=1 orange=1 disabled=0
+            [working]="isPublishing" (click)="togglePublish()">Unpublish</publish-button>
+          <publish-button *ngIf="editStatus" [model]="story" visible=1 red=1 disabled=0 working=0
+            (click)="confirmDelete()">Delete</publish-button>
+        </template>
+      </dd>
+
+      <dt *ngIf="isPublished && isScheduled">Publishing</dt>
+      <dt *ngIf="isPublished && !isScheduled">Published</dt>
+      <dd *ngIf="isPublished"><p>{{story.publishedAt | date:"short"}}</p></dd>
+
+      <dt>Saved</dt>
+      <dd *ngIf="!id"><p>Not Saved</p></dd>
+      <dd *ngIf="id"><p *ngIf="story?.updatedAt">{{story.updatedAt | date:"short"}}</p></dd>
+
+      <dt>Progress</dt>
+      <dd *ngIf="!id">
+        <p *ngIf="!normalInvalid">Ready to create</p>
+        <p *ngIf="normalInvalid" class="error">Unable to create</p>
+        <button *ngIf="normalInvalid" (click)="showProblems()" class="btn-link">{{normalInvalidCount}}</button>
+      </dd>
+      <dd *ngIf="id">
+        <template [ngIf]="strictInvalid">
+          <p *ngIf="isPublished || normalInvalid" class="error">Invalid episode</p>
+          <p *ngIf="notPublished && !normalInvalid">Not ready to publish</p>
+          <button (click)="showProblems()" class="btn-link">{{strictInvalidCount}}</button>
+        </template>
+        <template [ngIf]="notPublished && !strictInvalid">
+          <p *ngIf="changed">Ready after save</p>
+          <p *ngIf="!changed">Ready to publish</p>
+          <publish-button [model]="story" visible=1 orange=1 [disabled]="changed"
+            [working]="isPublishing" (click)="togglePublish()">Publish</publish-button>
+        </template>
+        <template [ngIf]="isPublished && !strictInvalid">
+          <p *ngIf="changed">Unsaved changes</p>
```

Comment:
Yeah, i'd like to see what folks think about that.  I didn't add it yet because I found it annoying to always have bold/colorful text there every time i edit something.